doc/BUILD.rst

```diff
@@ -0,0 +1,730 @@
+"""""""""""""""""""""""""""""""""""""
+
+* Default: ``true```
+* Options: ``true``, ``false``
+
+Controls whether or not Taisei can load game data (textures, shaders, etc) from
+``.zip`` files. Useful for distribution and packaging.
+
+**NOTE:** Setting this to ``false`` automatically disables ``-Dpackage_data``.
+
+.. code:: sh
+
+   meson configure build/ -Denable_zip=false
+
+In-Game Developer Options (``-Ddeveloper``)
+"""""""""""""""""""""""""""""""""""""""""""
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+For testing actual gameplay, you can set this option and it will enable cheats
+and other 'fast-forward' options by the pressing keys defined in
+``src/config.h``.
+
+.. code:: sh
+
+   meson configure build/ -Ddeveloper=true
+
+Build Type (``-Dbuildtype``)
+""""""""""""""""""""""""""""
+
+* Default: ``release``
+* Options: ``release``, ``debug``
+
+Sets the type of build. ``debug`` enables several additional debugging features,
+as well as reduced optimizations and more debugging symbols.
+
+.. code:: sh
+
+   meson configure build/ -Dbuildtype=debug
+
+Assertions (``-Db_ndebug``)
+"""""""""""""""""""""""""""
+
+* Default: ``true``
+* Options: ``true``, ``false``
+
+The name of this flag is opposite of what you'd expect. Think of it as "Not
+Debugging". It controls the ``NDEBUG`` declaration which is responsible for
+deactivating ``assert()`` functions.
+
+Setting to ``false`` will *enable* assertions (i.e: good for debugging).
+
+Keep ``true`` during release.
+
+.. code:: sh
+
+   meson configure build/ -Db_ndebug=false
+
+Strict Compiler Warnings (``-Dwerror``)
+"""""""""""""""""""""""""""""""""""""""
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+This option forces stricter checks against Taisei's codebase to ensure code
+health, treating all ``Warnings`` as ``Errors`` in the code.
+
+It's highly recommended to **enable** (i.e: ``true``) this whenever developing
+for the engine. Sometimes, it's overly-pedantic, but much of the time, it
+provides useful advice. (For example, it can detect potential null-pointer
+exceptions that may not be obvious to the human eye.)
+
+.. code:: sh
+
+   meson configure build/ -Dwerror=true
+
+Deprecation Warnings (``-Ddeprecation_warnings``)
+"""""""""""""""""""""""""""""""""""""""""""""""""
+
+* Default: ``default``
+* Options: ``error``, ``no-error``, ``ignore``, ``default``
+
+Sets deprecation warnings to either hard-fail (``error``), print as warnings but
+not trigger full errors if ``-Dwerror=true`` (``no-error``), and otherwise
+ignore them (``ignore``).
+
+Generally, ``no-error`` is the recommended default when using ``-Dwerror=true``.
+
+.. code:: sh
+
+   meson configure build/ -Ddeprecation_warnings=no-error
+
+
+Stack Trace Debugging (``-Db_sanitize``)
+""""""""""""""""""""""""""""""""""""""""
+
+This is useful for debugging crashes in the game. It uses
+`AddressSanitizer <https://github.com/google/sanitizers/wiki/AddressSanitizer>`__:
```

Comment:
I'd change this to something like:

```suggestion
Debugging with `Sanitizers <https://github.com/google/sanitizers/wiki>`__ (``-Db_sanitize``)
""""""""""""""""""""""""""""""""""""""""

This is useful for debugging memory management errors, leaks, and undefined behavior.
```

ASan does quite a lot more work than print stack traces.